elide-async/src/main/java/com/yahoo/elide/async/operation/AsyncQueryOperation.java

```diff
@@ -6,21 +6,51 @@
 package com.yahoo.elide.async.operation;
 
 import com.yahoo.elide.ElideResponse;
+import com.yahoo.elide.async.models.AsyncAPI;
+import com.yahoo.elide.async.models.AsyncAPIResult;
 import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
 import com.yahoo.elide.async.service.AsyncExecutorService;
+import com.yahoo.elide.core.RequestScope;
+import com.yahoo.elide.core.security.User;
 
 import org.apache.http.NoHttpResponseException;
 
 import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URISyntaxException;
+import java.util.Date;
 
 /**
  * AsyncQuery Execute Operation Interface.
  */
+@Slf4j
 public abstract class AsyncQueryOperation implements AsyncAPIOperation<AsyncQuery> {
     @Getter private AsyncExecutorService service;
+    private AsyncAPI queryObj;
+    private RequestScope scope;
 
-    public AsyncQueryOperation(AsyncExecutorService service) {
+    public AsyncQueryOperation(AsyncExecutorService service, AsyncAPI queryObj, RequestScope scope) {
         this.service = service;
+        this.queryObj = queryObj;
+        this.scope = scope;
+    }
+
+    @Override
+    public AsyncAPIResult call() throws URISyntaxException, NoHttpResponseException {
```

Comment:
We don't want to propagate these kinds of exceptions up the stack (it propagates a package dependency).